### PR TITLE
MDCT 292 - Remove autosave where not needed

### PR DESCRIPTION
**Pull request by ntsummers1**
# Description

Fixes [MDCT-292 - Remove autosave where not needed](https://qmacbis.atlassian.net/browse/MDCT-292). The autosave in the header should no longer show unless you're on a specific report that is meant to be autosaved. It also cleans up the interactions between webpages. Before we were not using react-router-dom's Links and were using regular a tags. While this provides for a smoother transition between pages, that might not be the intended outcome here. Please let me know if the a tags were intentional or if the new Link causes unexpected behavior.

## How to test

`./dev local`

Click around the app and checkout the various pages.

## Dependencies

None

# Get it done

Now that the PR is open this is what happens next

## Code authors checklist

- [x] I have performed a self-review of my code
- [x] I have added thorough tests. [Testing Doc](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research)
- [x] Necessary analytics were added, or no new analytics were required
- [ ] ~~I have made corresponding changes to the documentation~~
- [x] I have assigned the PR to myself

## Reviewers Checklist (Two different people)

- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

## Assignee

- [ ] I have closed the PR after the review and necessary changes (squashing preferred)

### CODE REVIEW

**Review by BearHanded — APPROVED** [review]

File: services/ui-src/src/AppRoutes.js [hunk 1]
```diff
@@ -36,10 +37,10 @@ const AppRoutes = () => {
       className={"App " + window.location.pathname.split("/")[1]}
       data-test="component-app"
     >
-      {VisibleHeader}
```
Comment (gmrabian) [inline services/ui-src/src/AppRoutes.js:39]: Header snuck back into the content - we still want to go by the change in [this PR](https://github.com/CMSgov/cms-carts-seds/pull/2483)?

File: services/ui-src/src/AppRoutes.js [hunk 2]
```diff
@@ -36,10 +37,10 @@ const AppRoutes = () => {
...
       <div className="app-content">
         <Spinner />
         <Router>
+          {VisibleHeader}
```
Comment (gmrabian) [inline services/ui-src/src/AppRoutes.js:43]: Header snuck back into the content - we still want to go by the change in [this PR](https://github.com/CMSgov/cms-carts-seds/pull/2483)?

**Comment by gmrabian** [comment]
Overall good, just that one comment

File: services/ui-src/src/AppRoutes.js (hunk 2 above)
Comment (ntsummers1) [reply services/ui-src/src/AppRoutes.js:43]: This is intentional on my end and I double checked with Brax this morning about this to make sure that was ok. The solution I went with here uses the Router's location prop to determine whether or not the autosave should be shown. I actually went with a different approach initially where I grabbed the pathname from the url instead [See the original commit here](https://github.com/CMSgov/cms-carts-seds/pull/2775/commits/0f12c18e72e300dc4634e79f02a32c52b3d2b18a). That original approach is fine, but it was a trade off of either using react-router-doms link implementation or using straight a tags. 

There is of course another option where we do a lot of state passup where we have the components being shown determine where its shown in the header themselves but that felt like way too much work.

**Review by juliuscaesar — APPROVED** [review]

**Pull request merged**